### ISSUE TITLE
Split logic

### ISSUE THREAD

**Issue by reinerRubin** [go-openapi/swag#28]
Hello,
I have been trying to fix split for "SomethingTTLSeconds" and rewritten the split function. But there is a problem I can not understand some test cases.

1) We have got three initialisms: "CAP", "WD", "CAPWD". A test sample is "CAPWDfolwdBylc".Why is it "Capwdfolwd Bylc" but not "CAP W Dfolwd Bylc"? Ok, we can not match CAPWD, but we can match CAP. Do we assume that it is kind of failed "CAPWD"? 

2) Should "HTTPo" be"HTT" + "Po" or just "HTTPo"?

**Comment by reinerRubin** [comment]
I have recreated this behavior, but found a little inconsistency in the current approach:
```
HTTPd => ([]string) (len=1 cap=10) {
 (string) (len=5) "HTTPd"
}
NOPEd => ([]string) (len=4 cap=10) {
 (string) (len=1) "N",
 (string) (len=1) "O",
 (string) (len=1) "P",
 (string) (len=2) "Ed"
}
```

I can keep this. But for me "HTTPd" should be also "H" + "T + "T" + "Pd". 

The first question seems more difficult for me. If you can formulate strict rules or some examples I'll add it to my implementation, add some tests and make PR.

**Comment by fredbi** [comment]
I found back the kind of behavior I wanted to avoid: https://github.com/go-swagger/go-swagger/issues/1202

**Comment by reinerRubin** [comment]
I found this usecase ("ELB.HTTPLoadBalancer") and my implementation cover it. I don't understand CAPWD samples. Why is "CAP" not counted as a initialisms

**Comment by fredbi** [comment]
I think the test cases are outdated in the light of your previous findings.
There are no clear cut rules for this kind of word processing. I would be inclined in favor of matching the longest initialism.

On the other hand, it is hard to follow in your PR the implications of your changes.

Perhaps a good yard stick would be to regenerate go-swagger examples with this version and figure out file names produces as well as linting the resulting code.

**Comment by reinerRubin** [comment]
Ok, I'll add some extra tests, check it with the go-swagger examples and try with my work project.

 In general I tried to remove regexes, make the split more clear and fix the bug ("SomethingTTLSeconds"). PR saves the logic with the longest initialism. So "HTTPS" would be in favor of "HTTP" and so on. But as I mentioned before "CAPWD" is not initialism here because of the next small letter, but "CAP" seems to be. 

Thanks for help! I'll try to do it on weekends.

**Comment by reinerRubin** [comment]
Ok, I have resolved the problem with IPv6. And now the PR saves the old behavior and fixes "SomethingTTLSeconds". All old swag tests are passed, except broken "CAPWD", but I have fixed these test cases. All go-swagger tests are passed too(with fixed "LinkLocalIPs" above).

I want to add more tests to swag, because some corner cases are present only in go-swagger, plus recheck all stuff I have done.
